File: src/main/java/com/example/life/game/grid/Border.java
package com.example.life.game.grid;

import com.example.life.game.cell.Position;

public final class Border {
    private Area area;

    public Border(Area area) {
        this.area = area;
    }

    public boolean isAt(Position position) {
        boolean isUpper = this.isUpper(position);
        boolean isLower = this.isLower(position);
        boolean isLeft = this.isLeft(position);
        boolean isRight = this.isRight(position);

        return (isUpper || isLower || isLeft || isRight);
    }

    private boolean isUpper(Position position) {
        return position.x() == -1 ? true : false;
    }

    private boolean isLower(Position position) {
        return position.x() == this.area.height() ? true : false;
    }

    private boolean isLeft(Position position) {
        return position.y() == -1 ? true : false;
    }

    private boolean isRight(Position position) {
        return position.y() == this.area.width() ? true : false;
    }
}

File: life/Grid/Border.cs
namespace Life
{
	interface IBorder
	{
		// Read methods.

		public bool IsAt(Position position);
	}

	public class Border : IBorder
	{
		private readonly Area area;

		public Border(Area area)
		{
			this.area = area;
		}

		public bool IsAt(Position position)
		{
			bool isUpper = this.isUpper(position);
			bool isLower = this.isLower(position);
			bool isLeft = this.isLeft(position);
			bool isRight = this.isRight(position);

			return (isUpper || isLower || isLeft || isRight);
		}

		private bool isUpper(Position position)
		{
			return position.X() == -1 ? true : false;
		}

		private bool isLower(Position position)
		{
			return position.X() == this.area.Height ? true : false;
		}

		private bool isLeft(Position position)
		{
			return position.Y() == -1 ? true : false;
		}

		private bool isRight(Position position)
		{
			return position.Y() == this.area.Width ? true : false;
		}
	}
}